IndexedDbGetAllEntries/explainer.md

```diff
@@ -0,0 +1,190 @@
+# IndexedDB: getAllEntries()
+
+## Author:
+- [Steve Becker](https://github.com/SteveBeckerMSFT)
+
+## Participate
+- https://github.com/w3c/IndexedDB/issues/206
+
+## Introduction
+
+[`IndexedDB`](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) is a transactional database for client-side storage.  Each record in the database contains a key-value pair.  [`getAll()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAll) enumerates database record values sorted by key in ascending order.  [`getAllKeys()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAllKeys) enumerates database record primary keys sorted by key in ascending order.
+
+This explainer proposes a new operation, `getAllEntries()`, which combines [`getAllKeys()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAllKeys) with [`getAll()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAll) to enumerate both primary keys and values at the same time.  For an [`IDBIndex`](https://developer.mozilla.org/en-US/docs/Web/API/IDBIndex), `getAllEntries()` also provides the record's index key in addition to the primary key and value.  Lastly, `getAllEntries()` offers a new option to enumerate records sorted by key in descending order.
+
+## WebIDL
+
+```js
+dictionary IDBGetAllEntriesOptions {
+  any query = null; // A key or an `IDBKeyRange` identifying the records to retrieve.
+  [EnforceRange] unsigned long count;
```

Comment:
Would it be helpful to add a short inline comment `// number of records to retrieve.`

Reply:
Good idea.  Done!